tests/modules/basics.c

```diff
@@ -963,6 +986,10 @@ int RedisModule_OnLoad(RedisModuleCtx *ctx, RedisModuleString **argv, int argc)
         TestStringAppend,"write deny-oom",1,1,1) == REDISMODULE_ERR)
         return REDISMODULE_ERR;
 
+    if (RedisModule_CreateCommand(ctx,"test.string.trim",
+        TestTrimString,"write deny-oom",1,1,1) == REDISMODULE_ERR)
+        return REDISMODULE_ERR;
```

Comment:
there's no need to create a command (we don't use it from the tcl test).
if you included the assertion in TestBasics that's enough for this one.

however, i don't see any test that checks RM_Call with an overly bloated string (calling SET command).
i.e. similar to the test we added for scripts, let's add one for modules.

Reply:
Here as well I followed the file's tests structure. and call the command at: `T("test.string.trim","");` for example `test.string.append.am`. 

I couldn't find a way to test the SET command because we increase the refcount of the given string object in 
`moduleCreateArgvFromUserFormat` and in `tryObjectEncoding` we don't trim if the object refcount is > 1.

Reply:
i'm sorry. i misread the test infra (the `T` macro). you do need to register the command.

ok, so because of the refcount issue you mentioned above, if a module generates a strings with some excess space, and passes it directly to RM_Call, it'll not get trimmed.
however, if the module passes an argument it got from the network forward to RM_Call, that one will get trimmed when RedisModuleCommandDispatcher exits, right?
so can we test that the same way we tested SET command in #11766 (i.e. by a module command that's a wrapper for SET)?

Reply:
Added